druntime/src/core/builtins.d

```diff
@@ -51,3 +51,41 @@ version (LDC)
 
 /// Writes `s` to `stderr` during CTFE (does nothing at runtime).
 void __ctfeWrite(scope const(char)[] s) @nogc @safe pure nothrow {}
+
+version (GNU)
+{
+    alias expect = __builtin_expect;
+    alias trap = __builtin_trap;
```

Comment:
Can public symbols always be documented ?
In this case, it's as simple as a link to https://gcc.gnu.org/onlinedocs/gcc/Other-Builtins.html#index-_005f_005fbuiltin_005fexpect